taf/utils.py

```diff
@@ -232,8 +234,47 @@ def safely_save_json_to_disk(data, permanent_path):
 
     temp_file_path = tfile.name
     tfile.close()
-    shutil.copy(temp_file_path, permanent_path)
-    os.remove(temp_file_path)
+    safely_move_file(temp_file_path, permanent_path, overwrite=True)
+
+
+def safely_move_file(src, dst, overwrite=False):
+    """Rename a file from ``src`` to ``dst``.
+
+    *   Moves must be atomic.  ``shutil.move()`` is not atomic.
+        Note that multiple threads may try to write to the cache at once,
+        so atomicity is required to ensure the serving on one thread doesn't
+        pick up a partially saved image from another thread.
+
+    *   Moves must work across filesystems.  Often temp directories and the
+        cache directories live on different filesystems.  ``os.rename()`` can
+        throw errors if run across filesystems.
+
+    So we try ``os.rename()``, but if we detect a cross-filesystem copy, we
+    switch to ``shutil.move()`` with some wrappers to make it atomic.
+    https://alexwlchan.net/2019/03/atomic-cross-filesystem-moves-in-python/
+    """
```

Comment:
I did get an error because cloned directory was on the C disk (in the temp folder) and the actual location was E:\\.... Cannot reply to other comments here, but I created both issues #169  and #170 . I tried to quickly add some logic for creating determining default branch, but that broke something, so won't rush it. If they need to create new repositories, we should hardcode main before giving this to them